wv2util/ExperimentalFeature.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace wv2util
+{
+    public class ExperimentalFeature : IEquatable<ExperimentalFeature>, IComparable<ExperimentalFeature>
+    {
+        public string Name { get; set; }
+        
+        public bool isEnabled;
+
+        public bool IsEnabled
+        {
+            get { return isEnabled; }
+            set
+            {
+                if (value)
+                {
+                    turnOn();
+                }
+                else
+                {
+                    turnOff();
+                }
+                isEnabled = value;
+            }
+        }
+
+        public string Description { get; set; }
+
+        private Action turnOn, turnOff;
+
+        public ExperimentalFeature(Action turnOn, Action turnOff, Func<bool> isOn)
+        {
+            this.turnOn = turnOn;
+            this.turnOff = turnOff;
+            isEnabled = isOn();
+        }
+
+        static public bool IsEnabledForApp(string commandLine, string envVars)
+        {
+            // TODO implement
+            return false;
+        }
+
+        public int CompareTo(ExperimentalFeature other)
+        {
+            return Name.CompareTo(other.Name);
+        }
+
+        public bool Equals(ExperimentalFeature other)
+        {
+            return Name.Equals(other.Name);
+        }
+    }
+
+    public class EnvVarExperimentalFeature : ExperimentalFeature
+    {
+        public EnvVarExperimentalFeature(string envVar, string onVal, string offVal) : base(
+            () =>
+            {
+                // Turn on:
+                Environment.SetEnvironmentVariable(envVar, onVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                // Turn off:
+                Environment.SetEnvironmentVariable(envVar, offVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                string val = Environment.GetEnvironmentVariable(envVar, EnvironmentVariableTarget.User);
+                return val != null && val == onVal;
+            })
+        {
+        }
+    }
+
+    public class ExperimentalFeatureList : ObservableCollection<ExperimentalFeature>
+    {
+        private List<ExperimentalFeature> m_List = new List<ExperimentalFeature>();
+
+        public ExperimentalFeatureList()
+        {
+            // Visual Hosting
+            Items.Add(new EnvVarExperimentalFeature(
+                "COREWEBVIEW2_FORCED_HOSTING_MODE",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_VISUAL",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_WINDOW")
+            {
+                Name = "Visual hosting",
+                Description = "When on apps will be running in Visual Hosting instead of regular Window hosting"
+            });
+
+            // Canary self-hosting
+            Items.Add(new EnvVarExperimentalFeature(
```

Comment:
Why use the environment variable to set this override versus a registry key?
Pros:
* overrides in the environment variables will have higher precedent than registry.

Cons:
* changes to the environment variables won't apply to processes that have already started, unlike changes to the registry.
* the mechanism we're switching to for selfhost uses registry (Vicky or Victor will know if its HKLM or HKCU) so using registry here will be more likely to reflect that actual state.

Reply:
_changes to the environment variables won't apply to processes that have already started, unlike changes to the registry._
If the process already started, nothing can change its runtime regardless of the method we choose to change this, right?

Reply:
I mean if the host app creates another webview2. The next webview2 created will respect the updated regkey but the updated env var isn't applied to the already running host app process.

Reply:
I understand that, I just thought that's an unlikely scenario, but I guess it can still happen.